### PR TITLE
Export Logging for Plugins

### PR DESCRIPTION
**Pull request by krishnaglick**
## Change description

Core will now log errors emitted from plugins.

### Development

- [X] Application changes have been tested appropriately

### Impact

- [X] Code follows company security practices and guidelines
- [X] Security impact of change has been considered
- [X] Performance impact of change has been considered
- [X] Possible migration needs considered (model migrations, config migrations, etc.)

### Code review

- [X] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [X] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: core/__tests__/models/destination/plugins/exportRecord.ts [hunk 1]
```diff
@@ -41,12 +46,16 @@ describe("models/destination - with custom exportRecord plugin", () => {
...
+  afterEach(() => {
+    mockLog.mockReset();
+  });
+  afterAll(() => {
+    logSpy.mockRestore();
   });
 
   beforeAll(async () => {
+    await helper.truncate();
+    ({ model } = await helper.factories.properties());
```
Comment (krishnaglick) [inline core/__tests__/models/destination/plugins/exportRecord.ts:58]: Moved this down.

File: core/src/modules/ops/destination.ts [hunk 2]
```diff
@@ -655,6 +655,17 @@ export namespace DestinationOps {
...
+  function logExportError(destination: Destination, error: Error) {
+    const recordId = error["recordId"]; // it might have one of these
+    log(
+      `ExportError  Destination: ${destination.id} - ${
+        destination.name
+      } - Record: ${recordId ?? "Unknown"}`,
+      "warning",
+      { error }
+    );
+  }
```
Comment (krishnaglick) [inline core/src/modules/ops/destination.ts:667]: Thank you Brian!

File: ui/ui-components/pages/model/[modelId]/record/[recordId]/edit.tsx [hunk 3]
```diff
@@ -237,9 +237,7 @@ export default function Page(props) {
               <LoadingButton
                 variant="outline-primary"
                 disabled={loading}
-                onClick={() => {
-                  exportRecord();
-                }}
+                onClick={exportRecord}
```
Comment (krishnaglick) [inline ui/ui-components/pages/model/[modelId]/record/[recordId]/edit.tsx:240]: Noticed this while searching for `exportRecord`

File: core/src/modules/ops/destination.ts [hunk 4]
```diff
@@ -663,6 +674,20 @@ export namespace DestinationOps {
...
+        for (const error of combinedError.errors) {
+          logExportError(destination, error);
+        }
+      }
+    }
+    if (exportResult?.errors) {
+      for (const error of exportResult?.errors) {
+        logExportError(destination, error);
+      }
+    }
```
Comment (evantahler) [inline core/src/modules/ops/destination.ts:689]: 👍 covered both the case with a provided `combinedError` and when `exportResult` has errors.

**Review by evantahler — APPROVED** [review]
👍 `updateExports()` seems like a good place to put this!

**Pull request merged**